providers/base/bin/check-ubuntu-desktop-recommends.sh

```diff
@@ -9,7 +9,7 @@ while read -r pkg; do
     if ! dpkg-query -W -f='${Status}\n' "$pkg" 2>&1 | grep "install ok installed" >/dev/null 2>&1 && [ "$pkg" != "libreoffice-ogltrans" ]; then
         noninstalled+=("$pkg")
     fi
-done < <(apt-cache show ubuntu-desktop | grep ^Recommends | head -n 1 | cut -d : -f 2- | xargs | sed 's/ //g' | tr , $'\n')
+done < <(apt-cache show ubuntu-desktop="$(dpkg-query --showformat='${Version}' --show ubuntu-desktop)" | grep ^Recommends | head -n 1 | cut -d : -f 2- | xargs | sed 's/ //g' | tr , $'\n')
```

Comment:
This line is starting to get really big. I would compute the version number first, then use it in the `apt-cache...` command.

Reply:
@pieqq would you please review it again?